index.d.ts

```diff
@@ -0,0 +1,61 @@
+import { FastifyPlugin } from "fastify";
+import {
+  Connection,
+  ConnectionOptions,
+  escape,
+  format,
+  Pool,
+  PoolOptions,
+} from "mysql2";
+import {
+  Connection as PromiseConnection,
+  Pool as PromisePool,
+} from "mysql2/promise";
+
+export type MySQLConnection = Pick<Connection, "query" | "execute"> & {
+  connection: Connection;
+  format: typeof format;
+  escape: typeof escape;
+};
+
+export type MySQLPool = Pick<Pool, "query" | "execute" | "getConnection"> & {
+  pool: Pool;
+  format: typeof format;
+  escape: typeof escape;
+};
+
+export type MySQLPromiseConnection = Pick<
+  PromiseConnection,
+  "query" | "execute"
+> & {
+  connection: PromiseConnection;
+  format: typeof format;
+  escape: typeof escape;
+};
+
+export type MySQLPromisePool = Pick<
+  PromisePool,
+  "query" | "execute" | "getConnection"
+> & {
+  pool: PromisePool;
+  format: typeof format;
+  escape: typeof escape;
+};
+
+declare module "fastify" {
+  interface FastifyInstance {
+    mysql: {};
```

Comment:
What's the purpose of the placeholder type here?

Reply:
It is used for test case only, because I need the `mysql` in fastify instance so I can cast it to other types for testing. As `mysql2` have four different type for the client. I do not have a proper typing for it.